src/main/java/com/flansmod/common/guns/ShootableType.java

```diff
@@ -78,7 +78,9 @@ public abstract class ShootableType extends InfoType {
     public boolean breaksGlass = false;
     public float ignoreArmorProbability = 0;
     public float ignoreArmorDamageFactor = 0;
-
+    
+    public float blockPenetrationModifier = FlansMod.masterBlockPenetrationModifier;
```

Comment:
This won't work correctly with syncing. The value here is assigned when types are created during game setup, so when the config sync happens the default will change, but it won't be read by anything. The way we've got around that before is something like:

```Java
private float blockPenetrationModifier = -1F;

public void read() {
    if (split[0].equals("Block..")) {
        blockPenetrationModifier = Float.praseFloat(split[1]);
    }
}

public float getBlockPenetrationModifier() {
    if (blockPenetrationModifier > 0) {
        return blockPenetrationModifier;
    } else {
        return FlansMod.defaultBlockPenetrationModifier
    }
}

```

Or go one step further, and put all the logic in ShootableType:
```Java

public float getBlockPenetration(Block block) {
     
}
```

But yeah, that needs to be fixed before this is merged. Don't feel bad about it, it's a really easy thing to miss - I've made it a few times too.

Reply:
Yup, this is resolved. Thanks :)